tiledb/sm/array/array.cc

```diff
@@ -1123,6 +1123,124 @@ void Array::non_empty_domain(void* domain, bool* is_empty) {
   }
 }
 
+void Array::non_empty_domain_from_name(
+    const char* name, void* domain, bool* is_empty) {
+  // Sanity check
+  if (name == nullptr) {
+    throw std::invalid_argument(
```

Comment:
This is only the right exception if the C API is checking whether `name` is a valid string, and in particular is not `nullptr`. So that check in the C API needs to be added.

See #4915 for a function that can be used for this; it should merge soon. In this case, in order to avoid changing the signature for `name` to something other than `char *`, the new function can be called and then `.data()` called on the return value, which will return the original pointer if it's valid.

Reply:
Considering this a blocking change. Once #4915 is merged, I'll rebase and update accordingly.

Reply:
>Once https://github.com/TileDB-Inc/TileDB/pull/4915 is merged

Now merged